### PR TITLE
partition, raid, format basic functionality

### PR DESCRIPTION
**Pull request by splaspood**
* cmd/root.go: Additional wrapper functions for cli argument handling
* cmd/*.go: Large amount of rework for consitency and DRY across cmds
* cmd/format_partition.go: Working functionality around mkfs.* and mkswap

### CODE REVIEW

**Review by JAORMX — APPROVED** [review]

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: goreleaser
+
+on:
+  push:
+    branches: ['main', 'cpr-0']
```
Comment (mmlb) [inline .github/workflows/lint.yml:5]: why limit to any branches at all? if you do
```
on:
  push
``` or w/e the exact syntax is should work too I think.

File: .goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      - "--label=org.opencontainers.image.title={{.ProjectName}}"
+      - "--label=org.opencontainers.image.revision={{.FullCommit}}"
+      - "--label=org.opencontainers.image.version={{.Version}}"
+  -
+    use: buildx
+    goos: linux
+    goarch: arm64
+    image_templates:
+    - "ghcr.io/metal-toolbox/{{.ProjectName}}:{{ .Tag }}-arm64v8"
+    - "ghcr.io/metal-toolbox/{{.ProjectName}}:latest-arm64v8"
```
Comment (mmlb) [inline .goreleaser.yml:61]: v8 is strange I don't think I've ever seen that

File: .goreleaser.yml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    build_flag_templates:
+      - "--platform=linux/arm64/v8"
+      - "--label=org.opencontainers.image.created={{.Date}}"
+      - "--label=org.opencontainers.image.title={{.ProjectName}}"
+      - "--label=org.opencontainers.image.revision={{.FullCommit}}"
+      - "--label=org.opencontainers.image.version={{.Version}}"
+
+docker_manifests:
+  # https://goreleaser.com/customization/docker_manifest/
+  - name_template: ghcr.io/metal-toolbox/{{.ProjectName}}:{{ .Tag }}
```
Comment (mmlb) [inline .goreleaser.yml:71]: kind of annoying that some of the templates are tight vs padded. `{{.Tag}}` probably to stay in line with all the other uses here.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,4 @@
+FROM scratch
+COPY vogelkop /vogelkop
+ENTRYPOINT ["/vogelkop"]
+CMD ["--help"]
```
Comment (mmlb) [inline Dockerfile:4]: new line at eol here

File: cmd/configure_raid.go [hunk 5]
```diff
@@ -66,13 +62,5 @@ func callLinuxSWRaid(array RaidArray) {
...
-	
-	logger.Debugw("running mdadm", "cmd", cmd)
-
-	if out, err := cmd.CombinedOutput(); err != nil {
-		logger.Debugf("%s\n", out)
-		logger.Fatalw("Failed to run mdadm",
-			"array", array, "cmd", cmd, "err", err, "stderr", out)
-	}
+	_, _ = callCommand("mdadm", cmd_args...)
 }
```
Comment (mmlb) [inline cmd/configure_raid.go:66]: eol

File: cmd/configure_raid.go [hunk 6]
```diff
@@ -66,13 +62,5 @@ func callLinuxSWRaid(array RaidArray) {
...
-	cmd := exec.Command("mdadm", cmd_args...)
-	
-	logger.Debugw("running mdadm", "cmd", cmd)
-
-	if out, err := cmd.CombinedOutput(); err != nil {
-		logger.Debugf("%s\n", out)
-		logger.Fatalw("Failed to run mdadm",
-			"array", array, "cmd", cmd, "err", err, "stderr", out)
-	}
+	_, _ = callCommand("mdadm", cmd_args...)
```
Comment (mmlb) [inline cmd/configure_raid.go:65]: this is tickling the spidey senses, why are you ignoring err return?

File: cmd/format_partition.go [hunk 7]
```diff
@@ -10,18 +12,70 @@ var (
...
 		},
 	}
 )
 
 func init() {
-	formatPartitionCmd.PersistentFlags().String("disk", "/dev/sda", "Disk containing partition")
-	formatPartitionCmd.PersistentFlags().Int("partition", 1, "Partition number to be formatted")
-	formatPartitionCmd.PersistentFlags().String("filesystem", "ext4", "Filesystem to be applied to the partition")
+	formatPartitionCmd.PersistentFlags().String("device", "", "Block device")
+	formatPartitionCmd.PersistentFlags().String("filesystem-device", "", "Filesystem Block device")
```
Comment (mmlb) [inline cmd/format_partition.go:37]: whats the difference between device and filesystem-device? Does device get partitioned and then filesystem-device is one of the partition's block device?

File: cmd/format_partition.go [hunk 8]
```diff
@@ -10,18 +12,70 @@ var (
...
+		_, _ = callCommand("mkfs." + partition.Format, mkfs_options...)
+	}
+
+	partition.UUID = getBlockDeviceUUID(partition.BlockDevice)
+}
+
+func getBlockDeviceUUID(device string) (uuid string) {
+	b_uuid, _ := callCommand("blkid", "-s", "UUID", "-o", "value", device)
+	return strings.TrimRight(string(b_uuid), "\n")
 }
```
Comment (mmlb) [inline cmd/format_partition.go:81]: ditto... I bet all the files

File: cmd/root.go [hunk 9]
```diff
@@ -4,7 +4,12 @@ import (
 	"go.uber.org/zap"
 	"github.com/spf13/cobra"
 
-	"vogelkop/internal"
+	"github.com/metal-toolbox/vogelkop/internal"
+
+	"strconv"
+	"os/exec"
+	"strings"
+	"path/filepath"
```
Comment (mmlb) [inline cmd/root.go:12]: this looks pretty weird, most go files are just 2 import blocks. The stdlib stuff at the top and everything else on a second block. `gofumpt` doesn't quite fix this yet I think though :(.

File: .github/workflows/lint.yml (hunk 1 above)
Comment (splaspood) [reply .github/workflows/lint.yml:5]: Was used for testing, will remove.

File: cmd/format_partition.go (hunk 7 above)
Comment (mmlb) [reply cmd/format_partition.go:37]: I'm a little bit out of the loop still here. How does this tool get used? Can everything be done in one invocation from unpartitioned block dev -> many fs on top of raid? If not then I don't see the reason for device and fs-device. If so it still feels like format doesn't really need to know about both the dev and the fs-dev.

**Review by mmlb — APPROVED** [review]

**Review by matoszz — APPROVED** [review]

**Pull request merged**